### PR TITLE
docs: Add serialization how-to

### PR DESCRIPTION
**Pull request by lukeblevins**
GitHub Issue (If applicable): #

## PR Type

What kind of change does this PR introduce?

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Tested code with current [supported SDKs](../README.md#supported)
- [ ] Docs have been added/updated which fit [documentation template](https://github.com/nventive/Uno/blob/master/doc/.feature-template.md). (for bug fixes / features)
- [ ] [Unit Tests and/or UI Tests](doc/articles/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] [Wasm UI Tests](doc/articles/working-with-the-samples-apps.md#running-the-webassembly-ui-tests-snapshots) are not showing unexpected any differences. Validate PR `Screenshots Compare Test Run` results.
- [ ] Contains **NO** breaking changes
- [ ] Updated the [Release Notes](https://github.com/nventive/Uno/tree/master/doc/ReleaseNotes)
- [ ] Associated with an issue (GitHub or internal)

## Other information

Internal Issue (If applicable):

### CODE REVIEW

File: doc/Extensions/Serialization/HowTo-Serialization.md [hunk 1]
```diff
@@ -0,0 +1,123 @@
+# How-To: Serialize and Deserialize JSON Data
+
+Accessing the serialized and deserialized representation of an object can be important for dynamic, data-rich applications. Uno.Extensions supports the [new](https://devblogs.microsoft.com/dotnet/try-the-new-system-text-json-source-generator) serialization technique powered by code generation, but you can optionally revert to the previous one which uses reflection.
```
Comment (Xiaoy312) [inline doc/Extensions/Serialization/HowTo-Serialization.md:3]:
```suggestion
Accessing the serialized and deserialized representation of an object can be important for dynamic, data-rich applications. Uno.Extensions supports the [new serialization technique](https://devblogs.microsoft.com/dotnet/try-the-new-system-text-json-source-generator) powered by code generation, but you can optionally revert to the previous one which uses reflection.
```

Comment (Xiaoy312) [reply doc/Extensions/Serialization/HowTo-Serialization.md:3]: or, just put the actual name in the link description

File: doc/Extensions/Serialization/HowTo-Serialization.md [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+
+## Step-by-steps
+
+### 1. Opt into Serialization
```
Comment (Xiaoy312) [inline doc/Extensions/Serialization/HowTo-Serialization.md:7]:
```suggestion
### 1. Opt-in Serialization
```

File: doc/Extensions/Serialization/HowTo-Serialization.md [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+
+        public Person(string name, int age, double height, double weight)
+        {
+            Name = name;
+            Age = age;
+            Height = height;
+            Weight = weight;
+        }
+        public string? Name { get; set; }
+
```
Comment (Xiaoy312) [inline doc/Extensions/Serialization/HowTo-Serialization.md:41]:
```suggestion
        
        public string? Name { get; set; }
```

File: doc/Extensions/Serialization/HowTo-Serialization.md [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+```csharp
+    private IHost Host { get; }
+    
+    public App()
+    {
+        Host = UnoHost
+            .CreateDefaultBuilder()
+            .UseSerialization(services => 
+                services
+                    .AddJsonTypeInfo(PersonContext.Default.Person)
```
Comment (Xiaoy312) [inline doc/Extensions/Serialization/HowTo-Serialization.md:69]: could save a line here:
```suggestion
            .UseSerialization(services => services
                .AddJsonTypeInfo(PersonContext.Default.Person)
```
same for other build patterns below

File: doc/Extensions/Serialization/HowTo-Serialization.md [hunk 5]
```diff
@@ -0,0 +1,123 @@
...
+Accessing the serialized and deserialized representation of an object can be important for dynamic, data-rich applications. Uno.Extensions supports the [new](https://devblogs.microsoft.com/dotnet/try-the-new-system-text-json-source-generator) serialization technique powered by code generation, but you can optionally revert to the previous one which uses reflection.
+
+## Step-by-steps
+
+### 1. Opt into Serialization
+
+* Call the `UseSerialization()` method to register a serializer that implements `ISerializer` with the service collection:
+
+```csharp
+    private IHost Host { get; }
```
Comment (Xiaoy312) [inline doc/Extensions/Serialization/HowTo-Serialization.md:12]: image

1. for code snippets belonging to a `*` bullet item, indent them 1 tier deeper than the bullet to align them
2. dont start with +1 indent level within the code snippets

\#2 collides with \#1, so you just need to adjust the indent of: \`\`\`

File: doc/Extensions/Serialization/HowTo-Serialization.md [hunk 6]
```diff
@@ -0,0 +1,123 @@
...
+                .AddSingleton(new JsonSerializerOptions { PropertyNameCaseInsensitive = true });
+        })
+        .Build();
+    // ........ //
+}
+```
+
+### 4. Serialize and deserialize JSON data
+
+* Add a constructor parameter for the `ISerializer<Person>` to a view model you registered:
```
Comment (Xiaoy312) [inline doc/Extensions/Serialization/HowTo-Serialization.md:103]:
```suggestion
* Obtain an instance of `ISerializer<Person>` in the view-model from DI:
```

File: doc/Extensions/Serialization/HowTo-Serialization.md [hunk 7]
```diff
@@ -0,0 +1,123 @@
...
+        private readonly ISerializer<Person> jsonSerializer;
+
+        public MainViewModel(ISerializer<Person> jsonSerializer)
+        {
+            this.jsonSerializer = jsonSerializer;
+        }
+    }
+```
+
+* To represent JSON data in both its serialized and deserialized form, you can now do the following:
```
Comment (Xiaoy312) [inline doc/Extensions/Serialization/HowTo-Serialization.md:117]:
```suggestion
* You can now serialize a `Person` object or deserialize from json like below:
```

File: doc/Extensions/Serialization/HowTo-Serialization.md [hunk 8]
```diff
@@ -0,0 +1,123 @@
...
+private IHost Host { get; }
+
+public App()
+{
+    Host = UnoHost
+        .CreateDefaultBuilder()
+        .UseSerialization(services => 
+            services
+                .AddJsonTypeInfo(PersonContext.Default.Person)
+        )
```
Comment (lukeblevins) [inline doc/Extensions/Serialization/HowTo-Serialization.md:90]:
```suggestion
        .UseSerialization(services => services
            .AddJsonTypeInfo(PersonContext.Default.Person)
        )
```

File: doc/Extensions/Serialization/HowTo-Serialization.md [hunk 9]
```diff
@@ -0,0 +1,121 @@
...
+        private readonly ISerializer<Person> jsonSerializer;
+
+        public MainViewModel(ISerializer<Person> jsonSerializer)
+        {
+            this.jsonSerializer = jsonSerializer;
+        }
+    }
+    ```
+
+* You can now serialize a `Person` object or deserialize from JSON:
```
Comment (agneszitte-nventive) [inline doc/Extensions/Serialization/HowTo-Serialization.md:115]:
```suggestion
* You can now serialize a `Person` object or deserialize it from JSON:
```

**Review by nickrandolph — APPROVED** [review]

**Pull request merged**